### DISCUSSION TITLE
ESP32-S3 Driving ST7701S Display Problem

### DISCUSSION THREAD

**Discussion by jerrylong-km** [moononournation/Arduino_GFX#480] [General]
Hi Guys
I am trying to use ESP32 driving 6" ST7701S Display.
This is the connection of ESP32 and Display
![1](https://github.com/moononournation/Arduino_GFX/assets/53472123/8c122040-78b4-4327-9128-85742cd13bc8)
This is the schematic
![2](https://github.com/moononournation/Arduino_GFX/assets/53472123/ff3d7278-e302-449e-9fd7-5740f11eefb3)

**Comment by jerrylong-km** [comment]
This is my testing code:
[ST7701S_LvglHelloWorld.zip](https://github.com/moononournation/Arduino_GFX/files/15443666/ST7701S_LvglHelloWorld.zip)
The BackLight is on when power-up, but shows nothing, could anyone help me to find out the problem? Thanks

**Comment by jerrylong-km** [comment]
My "Arduino_GFX_Library" version is 1.3.1
![3](https://github.com/moononournation/Arduino_GFX/assets/53472123/2d2607f9-14ea-4b53-9d4d-9f492413072b)
It seems that new version of GFX Lib does not support ST7701S anymore?

**Comment by jerrylong-km** [comment]
This is the document of the Display
[6寸长条屏.pdf](https://github.com/moononournation/Arduino_GFX/files/15443680/6.pdf)

**Comment by jerrylong-km** [comment]
I have used [Arduino_GFX_Library 1.4.7](https://github.com/moononournation/Arduino_GFX/tree/master/src) (latest version), and tested type of st7701 for type1 to type9, none of them works for my display.
![7](https://github.com/moononournation/Arduino_GFX/assets/53472123/473002b7-2745-446d-ba75-26e6d400c7b8)

![8](https://github.com/moononournation/Arduino_GFX/assets/53472123/4ca175bf-30ef-433b-9729-b80b0ae31ce9)

anythings else that I can try?

**Comment by showdown** [comment]
I was able to use the below code to get my ST7701 display working... kind of.  It's irregular but maybe you can build off this: 

```
#include <Arduino_GFX_Library.h>

Arduino_XCA9554SWSPI *expander = new Arduino_XCA9554SWSPI(
  PCA_TFT_RESET, PCA_TFT_CS, PCA_TFT_SCK, PCA_TFT_MOSI,
  &Wire, 0x3F);

Arduino_ESP32RGBPanel *rgbpanel = new Arduino_ESP32RGBPanel(
  TFT_DE, TFT_VSYNC, TFT_HSYNC, TFT_PCLK,
  TFT_R1, TFT_R2, TFT_R3, TFT_R4, TFT_R5,
  TFT_G0, TFT_G1, TFT_G2, TFT_G3, TFT_G4, TFT_G5,
  TFT_B1, TFT_B2, TFT_B3, TFT_B4, TFT_B5,
  1 /* hsync_polarity */, 50 /* hsync_front_porch */, 2 /* hsync_pulse_width */, 44 /* hsync_back_porch */,
  1 /* vsync_polarity */, 16 /* vsync_front_porch */, 2 /* vsync_pulse_width */, 18 /* vsync_back_porch */
);

Arduino_RGB_Display *gfx = new Arduino_RGB_Display(
  // 2.1" 480x480 round display adjust as needed
  480 /* width */, 480 /* height */, rgbpanel, 0 /* rotation */, true /* auto_flush */,
  expander, GFX_NOT_DEFINED /* RST */, TL021WVC02_init_operations, sizeof(TL021WVC02_init_operations));
```
